postgres and swagger added, first schema mvp

Euka-e opened the pull request:
<img width="1323" height="659" alt="image" src="https://github.com/user-attachments/assets/57eb0765-9b38-42ab-8164-8cbeb5d5e5a1" />
This pull request introduces several new dependencies and updates to support Prisma ORM and Swagger integration, as well as improvements to TypeScript linting rules. The most significant changes are the addition of Prisma and Swagger-related packages, new scripts for database management, and an update to the ESLint configuration to encourage safer TypeScript practices.

**Dependency and Tooling Additions**

* Added Prisma ORM and related tools for database management, including `@prisma/client` and `prisma`, as well as new scripts in `package.json` for generating, migrating, and resetting the database.
* Added Swagger integration for API documentation with `@nestjs/swagger` and `swagger-ui-express`, along with supporting packages such as `class-transformer` and `class-validator`.

**Configuration and Linting Updates**

* Updated ESLint configuration to change `@typescript-eslint/no-explicit-any` from "off" to "warn", encouraging developers to avoid using `any` type in TypeScript.

**Dependency Management**

* Updated `pnpm-lock.yaml` to reflect all new and updated dependencies, ensuring lockfile consistency with the new packages and their versions.

**Summary of Most Important Changes**

* **Prisma ORM Integration**
  - Added `@prisma/client` and `prisma` packages and related dev scripts for database schema management and development workflows.

* **Swagger and Validation Support**
  - Added `@nestjs/swagger`, `swagger-ui-express`, `class-transformer`, and `class-validator` to enable auto-generated API documentation and request validation.

* **Configuration Enhancements**
  - Added `@nestjs/config` for environment variable and configuration management.

* **Linting Improvements**
  - Changed ESLint rule for `@typescript-eslint/no-explicit-any` from "off" to "warn" to encourage better type safety.

* **Lockfile and Dependency Synchronization**
  - Updated `pnpm-lock.yaml` to include all new and updated dependencies, ensuring reproducible builds.

Euka-e committed:
postgres and swagger added, first schema mvp

Euka-e committed:
Add client-user module and enhance package tracking logic

Introduces the client-user module with controller, service, DTOs, and tests for managing user-client relationships. Refactors package and shipment logic: adds new fields and status enum to the Package model, updates Prisma schema and migration, and implements contextual package tracking endpoints and status updates. Also adds a Postman collection for API testing and updates related DTOs and controllers for improved package and shipment management.

neriheredia reviewed (changes requested):
# 💡 Propuesta: Interceptor + Helpers para Respuestas Unificadas

Estuve trabajando en una forma de **unificar todas las respuestas de la API** para que siempre salgan con el mismo formato `{ data, meta }`, y además simplificar el manejo de paginación. Les dejo el detalle de la idea 👇

---

## 🎯 Objetivos

- Todas las respuestas sigan un mismo shape:
  ```json
  {
    "data": [...],
    "meta": {
      "status": 200,
      "message": "Operación exitosa",
      "timestamp": "2025-09-20T00:00:00.000Z",
      "path": "/api/..."
    }
  }
  Soporte fácil para paginación (meta.pagination).

Posibilidad de configurar:

Mensaje de éxito (@ApiSuccessMessage)

Clave de colección distinta a items (@ApiCollectionKey).

Helpers para que los servicios devuelvan { items, pagination } sin repetir lógica.
Helper de paginación
 ```json
// src/common/utils/pagination.util.ts
export interface PaginationMeta {
  page: number;
  limit: number;
  total: number;
  pages: number;
}

export interface PaginatedResult<T> {
  items: T[];
  pagination: PaginationMeta;
}

export function buildPaginationMeta(total: number, page = 1, limit = 20): PaginationMeta {
  const safePage = Math.max(1, Number(page));
  const safeLimit = Math.max(1, Number(limit));
  return {
    page: safePage,
    limit: safeLimit,
    total,
    pages: Math.ceil(total / safeLimit),
  };
}

export function buildPaginatedResult<T>(
  items: T[],
  total: number,
  page?: number,
  limit?: number,
): PaginatedResult<T> {
  return {
    items,
    pagination: buildPaginationMeta(total, page, limit),
  };
}
 ```
Decoradores
 ```json
// src/common/decorators/response.decorator.ts
import { SetMetadata, createParamDecorator, ExecutionContext } from '@nestjs/common';

export const META_RESPONSE_MESSAGE = 'meta:response_message';
export const META_COLLECTION_KEY   = 'meta:collection_key';

export const ApiSuccessMessage = (message: string) =>
  SetMetadata(META_RESPONSE_MESSAGE, message);

export const ApiCollectionKey = (key: string) =>
  SetMetadata(META_COLLECTION_KEY, key);

export const PaginationQuery = createParamDecorator(
  (data: unknown, ctx: ExecutionContext) => {
    const req = ctx.switchToHttp().getRequest();
    const page  = Number(req.query.page ?? 1);
    const limit = Number(req.query.limit ?? 20);
    return { page, limit };
  },
);
 ```
Interceptor actualizado
 ```json
// src/common/interceptors/transform-response.interceptor.ts
import { CallHandler, ExecutionContext, NestInterceptor, Injectable } from '@nestjs/common';
import { Reflector } from '@nestjs/core';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { Response, Request } from 'express';
import { META_COLLECTION_KEY, META_RESPONSE_MESSAGE } from '../decorators/response.decorator';

@Injectable()
export class TransformResponseInterceptor<T extends object>
  implements NestInterceptor<T, any>
{
  constructor(private readonly reflector: Reflector) {}

  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
    const ctx = context.switchToHttp();
    const res = ctx.getResponse<Response>();
    const req = ctx.getRequest<Request>();

    const handler = context.getHandler();
    const controller = context.getClass();

    const message =
      this.reflector.get<string>(META_RESPONSE_MESSAGE, handler) ??
      this.reflector.get<string>(META_RESPONSE_MESSAGE, controller) ??
      'Operación exitosa';

    const collectionKey =
      this.reflector.get<string>(META_COLLECTION_KEY, handler) ??
      this.reflector.get<string>(META_COLLECTION_KEY, controller) ??
      'items';

    return next.handle().pipe(
      map((payload: any) => {
        const baseMeta = {
          status: res.statusCode,
          message,
          timestamp: new Date().toISOString(),
          path: req.url,
        };

        if (payload?.pagination && Array.isArray(payload.items)) {
          return { data: payload.items, meta: { ...baseMeta, pagination: payload.pagination } };
        }

        if (payload?.pagination && Array.isArray(payload[collectionKey])) {
          return { data: payload[collectionKey], meta: { ...baseMeta, pagination: payload.pagination } };
        }

        return { data: payload, meta: baseMeta };
      }),
    );
  }
}
 ```
Ejemplo en un Service
 ```json
// products.service.ts
import { buildPaginatedResult } from '@/common/utils/pagination.util';

async getAllPlatformProducts({ page = 1, limit = 20 }) {
  const skip = (page - 1) * limit;

  const [products, total] = await Promise.all([
    this.prisma.product.findMany({ skip, take: limit }),
    this.prisma.product.count(),
  ]);

  return buildPaginatedResult(products, total, page, limit);
}
 ```
Ejemplo en un Controller

 ```json
Caso estándar
@Get()
@ApiSuccessMessage('Listado de productos obtenido')
async list(@Query() query) {
  return this.productsService.getAllPlatformProducts(query);
}
 ```
 ```json

Caso con clave custom
@Get('platform')
@ApiSuccessMessage('Productos de plataforma')
@ApiCollectionKey('products') // <- el servicio devuelve { products, pagination }
async platform(@Query() query) {
  return this.productsService.getAllPlatformProducts(query);
}
 ```

✅ Beneficios

Respuestas consistentes en toda la API.

Servicios más simples: devuelven { items, pagination } o { <clave>, pagination }.

Control granular de:

Mensajes (@ApiSuccessMessage)

Nombre de la colección (@ApiCollectionKey)

Menos boilerplate en controladores y servicios.

Euka-e committed:
Add unified response interceptor and pagination utils

Introduces a global TransformResponseInterceptor to standardize API responses and success messages using new decorators. Refactors pagination logic across services to use a shared utility, updates DTOs and controller signatures for paginated endpoints, and improves type safety. Removes legacy test files and enhances OpenAPI documentation for paginated endpoints.
